Index: org/liblouis/liblouisutdml.java
===================================================================
--- org/liblouis/liblouisutdml.java
+++ org/liblouis/liblouisutdml.java
@@ -80,15 +80,13 @@
   public static native void initialize (String dataPath, String 
   writeablePath, String logFile);
   
-  public static void loadLibrary (String libraryPath, String 
-  librarySuffix) throws Exception {
+  public static void loadLibrary (String liblouisPath, String liblouisutdmlPath) throws Exception {
  if (libraryLoaded)
       return;
-    if (libraryPath == null || librarySuffix == null)
-      throw new Exception (
-      "Could not load libraries. libraryPath or librarySuffix undefined.");
-    System.load (libraryPath + "/liblouis" + librarySuffix);
-    System.load (libraryPath + "/liblouisutdml" + librarySuffix);
+    if (liblouisPath == null || liblouisutdmlPath == null)
+      throw new Exception ("Could not load libraries. liblouisPath or liblouisutdmlPath undefined.");
+    System.load (liblouisPath);
+    System.load (liblouisutdmlPath);
     libraryLoaded = true;
   }
   
